src/privacy_analysis/packet_analysis/packet_privacy_port.py

```diff
@@ -0,0 +1,45 @@
+#! /usr/bin/env python3
+from src.privacy_analysis.packet_analysis.packet_privacy import PacketPrivacy
+from scapy.all import TCP, UDP
+from scapy.all import Packet
+from dashboard.alerts.alert import alert, ALERT_TYPE, SEVERITY
+
+suspicious_ports = [21, 22, 23, 2323, 25, 110, 111, 135]
+
+
+# TODO: separate sus ports for TCP/UDP?
+
+class PacketPrivacyPort(PacketPrivacy):
+    def __call__(self, packet: Packet):
+        is_TCP = packet.haslayer(TCP)
+        is_UDP = packet.haslayer(UDP)
+
+        # Perform TCP and UDP checks
+        if is_TCP or is_UDP:
+            proto_type = TCP if is_TCP else UDP
+
+            # Scan for using port 80 and the plaintext for privacy leaks
+            if (packet[proto_type].dport == 80) or (packet[proto_type].sport == 80):
+                alert_port_80 = alert("Sending data over unencrypted port.", ALERT_TYPE.PRIVACY, SEVERITY.ALERT)
+                alert_port_80.alert()
+                self.__scan_plaintext(packet)
+
+            # Monitor suspicious ports
+            print("Monitoring suspicious ports")
+            if packet[proto_type].dport in suspicious_ports:
```

Comment:
Why isn't the source port being checked here as well?

Reply:
So most systems use the dynamic port range (49152-65535) for outgoing traffic and they are usually chosen at random since they are unassigned ports. There should not be a need to check the source port.